stream_alert/rule_processor/handler.py

```diff
@@ -167,68 +171,141 @@ def get_alerts(self):
         """
         return self._alerts
 
+    @staticmethod
+    def _segment_records_by_count(record_list, max_count):
+        """Segment records by length
+
+        Args:
+            record_list (list): The original records list to be segmented
+            max_count (int): The max amount of records to yield per group
+        """
+        for item in range(0, len(record_list), max_count):
+            yield record_list[item:item + max_count]
+
+    def _segment_records_by_size(self, record_batch):
+        """Segment record groups by size
+
+        Args:
+            record_batch (list): The original record batch to measure and segment
+
+        Returns:
+            generator: Used to iterate on each newly segmented group
+        """
+        split_factor = 1
+        len_batch = len(record_batch)
+
+        while True:
+            if len(str(record_batch[:len_batch / split_factor])) > MAX_BATCH_SIZE:
+                split_factor += 1
+            else:
+                break
+
+        return self._segment_records_by_count(record_batch, len_batch / split_factor)
+
+    @staticmethod
+    def _limit_record_size(batch):
+        """Limit the record size to be sent to Firehose
+
+        Args:
+            batch (list): Record batch to iterate on
+        """
+        for index, record in enumerate(batch):
+            if len(str(record)) > MAX_RECORD_SIZE:
```

Comment:
What is the type of `record`? If it's a dictionary, than `str(record)` will include whitespace between keys and values, which you might not have in the actual data you send over the wire

Reply:
@austinbyers it's a `dict`, what would you suggest in this case then?

Reply:
Since `str` will take up more space, it's definitely safe as is. If you want to pack in more data, you could `json.dumps` to calculate its size as if you were about to send it. I don't imagine `json.dumps` takes that much more time than `str`

Reply:
Just timed it, `json.dumps` is faster

Reply:
Cool! Good to know